File: Top-Interview-Questions-Leetcode/Python/Easy-Collections/Stock-Buy-and-Sell-2.py
from typing import List

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_profit, n = 0, len(prices)
        
        for i in range(1, n):
            if prices[i] > prices[i - 1]:
                max_profit += prices[i] - prices[i - 1]
        
        return max_profit

# Example usage
def main():
    prices = [7, 1, 5, 3, 6, 4]
    solution = Solution()
    result = solution.maxProfit(prices)
    print(f"Max profit: {result}")

if __name__ == "__main__":
    main()

File: Top-Interview-Questions-Leetcode/Easy-Collections/Array/Stock-Buy-and-Sell-2.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    int maxProfit(std::vector<int>& prices) {
        int max_profit = 0, n = prices.size();
        
        for (int i = 1; i < n; ++i) {
            if (prices[i] > prices[i - 1]) max_profit += prices[i] - prices[i - 1];
        }
        
        return max_profit;
    }
};

// Example usage
int main() {
    Solution solution;
    std::vector<int> prices = {7, 1, 5, 3, 6, 4};
    int result = solution.maxProfit(prices);
    std::cout << "Maximum profit: " << result << std::endl; // Output: 7
    return 0;
}